pkg/manager/manager.go

```diff
@@ -318,6 +384,88 @@ func (m *manager) DetachRepresentor(conf *types.PluginConf) error {
 		log.Info().Msgf("Restoring MTU %d on rep %s", conf.OrigRepState.MTU, conf.Representor)
 	}
 
+	// remove vlan config from uplink if configured to do so
+	if conf.SetUplinkVlan {
+		if err = m.deleteUplinkVlans(rep, conf); err != nil {
+			log.Warn().Msgf("Failed to delete trunk VLANs to parent uplink %v", err)
+		}
+	}
+
 	log.Info().Msgf("Detaching rep %s from the bridge %s", conf.Representor, conf.ActualBridge)
 	return m.nLink.LinkSetNoMaster(rep)
 }
+
+func (m *manager) deleteUplinkVlans(rep netlink.Link, conf *types.PluginConf) error {
+	var uplink netlink.Link
+	var err error
+
+	uplink, err = m.getPFUplinkOrBond(conf.PFName)
+	if err != nil {
+		return fmt.Errorf("failed to lookup PF for VF-Representor - PFName:%s: VF:%s %v",
+			conf.PFName, conf.Representor, err)
+	}
+
+	var vlans []int
+	if len(conf.Trunk) > 0 {
+		vlans = conf.Trunk
+	}
+
+	if conf.Vlan > 0 {
+		vlans = append(vlans, conf.Vlan)
+	}
+
+	var bridgeLink netlink.Link
+	bridgeLink, err = m.nLink.LinkByIndex(uplink.Attrs().MasterIndex)
+	if err != nil {
+		return fmt.Errorf("failed to lookup bridge index for interface:%s: %d %v",
+			uplink.Attrs().Name, uplink.Attrs().MasterIndex, err)
+	}
+
+	var currentbrif []string
+	currentbrif, err = utils.GetBridgeInterfaces(bridgeLink.Attrs().Name)
+	if err != nil {
+		return fmt.Errorf("failed to get bridge interfaces:%s: %v",
+			bridgeLink.Attrs().Name, err)
+	}
+
+	var allbrif map[int32][]*nl.BridgeVlanInfo
+	allbrif, _ = utils.BridgeVlanList(m.nLink)
+
+	var delvlans []int
+
+	for _, vlan := range vlans {
+		found := false
+
+	foundvlan:
+		for _, brif := range currentbrif {
+			brlink, iferr := m.nLink.LinkByName(brif)
+			if iferr != nil {
+				log.Warn().Msgf("could not lookup netlink info for %s: %v", brif, iferr)
+				continue
+			}
+
+			// skip the interface already being removed or the uplink, it obviously has these vlans.
+			// we are looking for any others
+			if rep.Attrs().Index == brlink.Attrs().Index || uplink.Attrs().Index == brlink.Attrs().Index {
```

Comment:
technically you could first remove VF rep from bridge and then attempt to delete vlans from uplink which would save a check here.

by no means a must.

Reply:
done